File: statsig-dotnet/src/Statsig/EvaluationDetails.cs
using Newtonsoft.Json;

namespace Statsig
{
    public class EvaluationDetails
    {
        [JsonProperty("lcut")] public long? Lcut = null;
        [JsonProperty("received_at")] public long? ReceivedAt = null;
        [JsonProperty("reason")] public string Reason = "";

        public EvaluationDetails(long? lcut, long? receivedAt, string reason)
        {
            Lcut = lcut;
            ReceivedAt = receivedAt;
            Reason = reason;
        }
    }
}

File: statsig-elixir/lib/statsig/evaluation_details.ex
defmodule Statsig.EvaluationDetails do
  defstruct [
    :reason,
    :lcut,
    :received_at,
    :version
  ]

  @type t :: %__MODULE__{
          reason: String.t(),
          lcut: non_neg_integer() | nil,
          received_at: non_neg_integer() | nil,
          version: non_neg_integer() | nil
        }
end
